core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -255,36 +262,42 @@ public static Socket connect(Properties props, String unixPathSuffix)
     final List<String> ipTypes = listIpTypes(props.getProperty("ipTypes", DEFAULT_IP_TYPES));
     logger.info(
         String.format("Connecting to Cloud SQL instance [%s] via SSL socket.", csqlInstanceName));
-    return getInstance().createSslSocket(csqlInstanceName, ipTypes, enableIamAuth);
+    return getInstance(props).createSslSocket(csqlInstanceName, ipTypes, enableIamAuth, serviceAccounts);
   }
 
   /**
    * Returns data that can be used to establish Cloud SQL SSL connection.
    */
-  public static SslData getSslData(String csqlInstanceName, boolean enableIamAuth)
+  public static SslData getSslData(String csqlInstanceName, boolean enableIamAuth, @Nullable List<String> serviceAccounts)
       throws IOException {
-    return getInstance().getCloudSqlInstance(csqlInstanceName, enableIamAuth).getSslData();
+    return getInstance(createPropertyForServiceImpersonation(serviceAccounts)).getCloudSqlInstance(csqlInstanceName, enableIamAuth, serviceAccounts).getSslData();
+  }
+
+  private static Properties createPropertyForServiceImpersonation(List<String> accounts) {
+    Properties props = new Properties();
+    props.setProperty(SERVICE_ACCOUNT_IMPERSONATION_PROPERTY, Joiner.on(",").join(accounts));
+    return props;
   }
 
   /**
    * Returns data that can be used to establish Cloud SQL SSL connection.
    */
   public static SslData getSslData(String csqlInstanceName) throws IOException {
-    return getSslData(csqlInstanceName, false);
+    return getSslData(csqlInstanceName, false, Collections.emptyList());
   }
 
   /**
    * Returns default ip address that can be used to establish Cloud SQL connection.
    */
   public static String getHostIp(String csqlInstanceName) throws IOException {
-    return getInstance().getHostIp(csqlInstanceName, listIpTypes(DEFAULT_IP_TYPES));
+    return getInstance(null).getHostIp(csqlInstanceName, listIpTypes(DEFAULT_IP_TYPES));
```

Comment:
These methods don't take enableIAMAuth so that means service impersonation couldn't even be used.